src/components/Model3DViewer/Model3DViewer.tsx

```diff
@@ -0,0 +1,202 @@
+import 'regenerator-runtime';
+
+import { Cognite3DModel, Cognite3DViewer, THREE } from '@cognite/3d-viewer';
+import * as sdk from '@cognite/sdk';
+import React from 'react';
+import { CacheObject, Callback, MouseScreenPosition } from '../../interfaces';
+import {
+  createViewer as originalCreateViewer,
+  parseBoundingBox,
+  setCameraPosition,
+  ViewerEventTypes,
+} from '../../utils';
+
+let createViewer = originalCreateViewer;
+
+type ClickHandler = (position: MouseScreenPosition) => void;
+
+export interface Model3DViewerProps {
+  boundingBox: THREE.Box3;
+  cache: CacheObject;
+  defaultCameraPosition?: boolean;
+  modelId: number;
+  onError?: Callback;
+  onProgress?: Callback;
+  onComplete?: Callback;
+  onReady?: Callback;
+  onClick?: Callback;
+  onCameraChange?: Callback;
+  projectName: string;
+  revisionId: number;
+}
+
+export function mockCreateViewer(mockFunction: any) {
+  createViewer = mockFunction || originalCreateViewer;
+}
+
+export class Model3DViewer extends React.Component<Model3DViewerProps> {
+  static defaultProps = {
+    boundingBox: parseBoundingBox({
+      min: [Infinity, Infinity, Infinity],
+      max: [-Infinity, -Infinity, -Infinity],
+    }),
+    defaultCameraPosition: true,
```

Comment:
After giving it some thoughts I would prefer `useDefaultCameraPosition` since `defaultCameraPosition` sound like a position I can set as default

Reply:
Agreed